### ISSUE TITLE
Is there a plan to split the cli and api?

### ISSUE THREAD

**Issue by mdebbar** [facebook/jscodeshift#179]
Many times, I found myself needing `jscodeshift`'s ability to traverse files with a pool of workers. But I don't always need the `jscodeshift` API for finding/replacing AST nodes.

It would be great to split the two, and provide a lightweight cli that only traverses files and uses pool of workers to run arbitrary code on those files.

Any thoughts on this?

**Comment by fkling** [comment]
There are no plans for this atm.

> I don't always need the `jscodeshift` API for finding/replacing nodes.

You don't have to use it. Is there a noticeable performance difference when the jscodeshift API is not loaded?

**Comment by mdebbar** [comment]
I mean there are many other use cases that need traversing files and run some code on each file (like `array.forEach()` but for files). For example: linting, collecting metrics, doing static analysis, etc. In those cases, it doesn't make sense to include the whole jscodeshift API (and its dependencies e.g. babel, babylon, recast, flow-parser, etc).

It works fine as it is now, but it's not ideal. If someone publishes a tool that uses `jscodeshift`'s file traversal, then it will unnecessarily include a lot of extra dependencies.

Basically, what I'm saying is I love `jscodeshift`'s file traversal and would like to use it for things other than code transformations 😃

**Comment by fkling** [comment]
That makes sense to me (and I'm glad you like the runner 😃 ). I don't think I have time to work on this myself, but if someone else wants to give it a try I'm happy to review the PR.
